packages/@sanity/components/src/snackbar/story.js

```diff
@@ -1,33 +1,145 @@
+/* eslint-disable react/no-multi-comp */
 import React from 'react'
-import Snackbar from 'part:@sanity/components/snackbar/default'
-import {storiesOf, action} from 'part:@sanity/storybook'
-import {withKnobs, select, text, number} from 'part:@sanity/storybook/addons/knobs'
+import PropTypes from 'prop-types'
+import Snackbar from 'part:@sanity/components/snackbar/item'
+import {storiesOf, addDecorator} from 'part:@sanity/storybook'
+import {withKnobs, select, text, button, boolean, number} from 'part:@sanity/storybook/addons/knobs'
 import Sanity from 'part:@sanity/storybook/addons/sanity'
+import SnackbarProvider from 'part:@sanity/components/snackbar/provider'
 
-const getKinds = () => select('Kind', ['success', 'error', 'warning', 'info'])
+function action(something) {
+  return () => console.log('action:', something)
+}
+const Provider = storyFn => (
+  <Sanity part="part:@sanity/components/snackbar/item" propTables={[Snackbar]}>
+    <SnackbarProvider>{storyFn()}</SnackbarProvider>
+  </Sanity>
+)
 
-storiesOf('Snackbar')
-  .addDecorator(withKnobs)
-  .add('Snackbar', () => (
-    <Sanity part="part:@sanity/components/snackbar/default" propTables={[Snackbar]}>
-      <Snackbar kind={getKinds()} timeout={number('timeout after (sec)', 500, 'props')}>
-        {text('content', 'This is the content')}
-      </Snackbar>
-    </Sanity>
+const globalDefaults = {
+  offset: 10,
+  isOpen: true,
+  id: new Date().getTime() + Math.floor(Math.random()),
+  setFocus: false,
+  onClose: action('onClose(), callback: onDismiss'),
+  onDismiss: action('onDismiss()')
+}
+
+class SnackQueue extends React.PureComponent {
+  static propTypes = {
+    snack: PropTypes.object
+  }
+  static contextTypes = {
+    addToSnackQueue: PropTypes.func
+  }
+
+  addToQueue = () => {
+    const {snack} = this.props
+    const newSnack = {
+      ...snack,
+      id: new Date().getTime() + Math.floor(Math.random())
+    }
+    this.context.addToSnackQueue(newSnack)
+  }
+
+  render() {
+    button('Add to queue', this.addToQueue, 'test')
+    return <div />
+  }
+}
+
+addDecorator(Provider)
+addDecorator(withKnobs)
+storiesOf('Snackbar', module)
+  .add('Default', () => (
+    <Snackbar
+      {...globalDefaults}
+      kind={select('Kind', ['success', 'error', 'warning', 'info'], 'info', 'props')}
+      title={text('Title', 'This is a title placeholder', 'props')}
+      subtitle={text('Subtitle', '', 'props')}
+      icon={boolean('With icons', false)}
+      isCloseable={boolean('isCloseable', false, 'props')}
+      action={{title: text('Action title', ''), callback: action('')}}
```

Comment:
The action should get a title here, so we can se inn the console that it is the callback action that is fired